tests/system/src/test/java/test/robot/javafx/scene/RobotTest.java

```diff
@@ -233,27 +237,50 @@ public void testScreenCapture() throws Exception {
         Pane pane = new StackPane();
         InvalidationListener invalidationListener = observable -> setSceneLatch.countDown();
         Platform.runLater(() -> {
-            pane.setBackground(new Background(new BackgroundFill(Color.MAGENTA, null, null)));
-            scene = new Scene(pane, SIZE, SIZE);
+            pane.setBackground(new Background(new BackgroundFill(MAGENTA, null, null)));
+            scene = new Scene(pane, WIDTH, HEIGHT);
             stage.sceneProperty().addListener(observable -> {
                 setSceneLatch.countDown();
                 stage.sceneProperty().removeListener(invalidationListener);
             });
             stage.setScene(scene);
         });
         waitForLatch(setSceneLatch, 5, "Timeout while waiting for scene to be set on stage.");
-        AtomicReference<WritableImage> screenCapture = new AtomicReference<>();
+        AtomicReference<WritableImage> screenCaptureNotScaledToFit = new AtomicReference<>();
+        AtomicReference<WritableImage> screenCaptureScaledToFit = new AtomicReference<>();
         Thread.sleep(1000);
         Platform.runLater(() -> {
             int x = (int) stage.getX();
             int y = (int) stage.getY();
-            screenCapture.set(robot.getScreenCapture(null, x, y, SIZE, SIZE, false));
+            screenCaptureNotScaledToFit.set(robot.getScreenCapture(null, x, y, WIDTH, HEIGHT, false));
+            screenCaptureScaledToFit.set(robot.getScreenCapture(null, x, y, WIDTH, HEIGHT, true));
             screenCaptureLatch.countDown();
         });
         waitForLatch(screenCaptureLatch, 5, "Timeout while waiting for robot.getScreenCapture().");
-        Assert.assertEquals(screenCapture.get().getWidth(), (double) SIZE, 0.0001);
-        Assert.assertEquals(screenCapture.get().getHeight(), (double) SIZE, 0.0001);
-        assertColorEquals(Color.MAGENTA, screenCapture.get().getPixelReader().getColor(150, 50), TOLERANCE);
+        double screenScaleX = Screen.getPrimary().getOutputScaleX();
+        double screenScaleY = Screen.getPrimary().getOutputScaleY();
+
+        // Should be scaled to the primary screen x and y scales. Note that screenCaptureScaledToFit and
+        // screenCaptureNotScaledToFit will be the same if screenScaleX = screenScaleY = 1.0 and in that case
+        // this is redundant.
+        Assert.assertEquals((double) WIDTH * screenScaleX,
+                screenCaptureNotScaledToFit.get().getWidth() * screenScaleX, 0.0001);
```

Comment:
The actual value is expected to already be scaled, so should not be multiplied by screenScaleX again. Both Mac and Windows in HiDPI fail this assertion. It should be:

```
        Assert.assertEquals((double) WIDTH * screenScaleX,
                screenCaptureNotScaledToFit.get().getWidth(), 0.0001);
```

same for height.

Reply:
Oh yes, of course. Thanks.